django/db/backends/base/client.py

```diff
@@ -20,7 +20,6 @@ def settings_to_cmd_args_env(cls, settings_dict, parameters):
         )
 
     def runshell(self, parameters):
-        args, env = self.settings_to_cmd_args_env(self.connection.settings_dict, parameters)
-        if env:
-            env = {**os.environ, **env}
+        args, extra_env = self.settings_to_cmd_args_env(self.connection.settings_dict, parameters)
+        env = {**os.environ, **extra_env} if extra_env else None
```

Comment:
I'd just stick with `env` over `extra_env` which will keep the diff to a minimum:
```suggestion
        args, env = self.settings_to_cmd_args_env(self.connection.settings_dict, parameters)
        env = {**os.environ, **env} if env else None
```